api/apps/v1alpha1/nemo_customizer_types.go

```diff
@@ -0,0 +1,925 @@
+	}
+}
+
+// GetStandardEnv returns the standard set of env variables for the NemoCustomizer container
+func (n *NemoCustomizer) GetStandardEnv() []corev1.EnvVar {
+	envVars := []corev1.EnvVar{
+		{
+			Name:  "NAMESPACE",
+			Value: n.Namespace,
+		},
+		{
+			Name: "HOST_IP",
+			ValueFrom: &corev1.EnvVarSource{
+				FieldRef: &corev1.ObjectFieldSelector{
+					APIVersion: "v1",
+					FieldPath:  "status.hostIP",
+				},
+			},
+		},
+		{
+			Name:  "CONFIG_PATH",
+			Value: "/app/config/config.yaml",
+		},
+		{
+			Name:  "CUSTOMIZATIONS_CALLBACK_URL",
+			Value: fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", n.GetName(), n.GetNamespace(), n.GetInternalServicePort()),
+		},
+		{
+			Name:  "LOG_LEVEL",
+			Value: "INFO",
+		},
+		{
+			Name: "WANDB_ENCRYPTION_KEY",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: &corev1.SecretKeySelector{
+					Key: n.Spec.WandBSecret.APIKeyKey,
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: n.Spec.WandBSecret.Name,
+					},
+				},
+			},
+		},
+		{
+			Name:  "WANDB_SECRET_NAME",
+			Value: n.Spec.WandBSecret.Name,
+		},
+		{
+			Name:  "WANDB_SECRET_KEY",
+			Value: n.Spec.WandBSecret.APIKeyKey,
+		},
+	}
+
+	// Append the environment variables for Postgres
+	envVars = append(envVars, n.GetPostgresEnv()...)
+
+	// Append the environment variables for OTel
+	if n.IsOtelEnabled() {
+		envVars = append(envVars, n.GetOtelEnv()...)
+	}
+
+	return envVars
+}
+
+// IsOtelEnabled returns true if Open Telemetry Collector is enabled
+func (n *NemoCustomizer) IsOtelEnabled() bool {
+	return n.Spec.OpenTelemetry.Enabled != nil && *n.Spec.OpenTelemetry.Enabled
+}
+
+// GetOtelEnv generates OpenTelemetry-related environment variables.
+func (n *NemoCustomizer) GetOtelEnv() []corev1.EnvVar {
+	var otelEnvVars []corev1.EnvVar
+
+	otelEnvVars = append(otelEnvVars,
+		corev1.EnvVar{Name: "OTEL_EXPORTER_OTLP_ENDPOINT", Value: n.Spec.OpenTelemetry.ExporterOtlpEndpoint},
+		corev1.EnvVar{Name: "OTEL_TRACES_EXPORTER", Value: n.Spec.OpenTelemetry.ExporterConfig.TracesExporter},
+		corev1.EnvVar{Name: "OTEL_METRICS_EXPORTER", Value: n.Spec.OpenTelemetry.ExporterConfig.MetricsExporter},
+		corev1.EnvVar{Name: "OTEL_LOGS_EXPORTER", Value: n.Spec.OpenTelemetry.ExporterConfig.LogsExporter},
+		corev1.EnvVar{Name: "OTEL_LOG_LEVEL", Value: n.Spec.OpenTelemetry.LogLevel},
+	)
+
+	if len(n.Spec.OpenTelemetry.ExcludedUrls) > 0 {
+		otelEnvVars = append(otelEnvVars, corev1.EnvVar{
+			Name:  "OTEL_PYTHON_EXCLUDED_URLS",
+			Value: strings.Join(n.Spec.OpenTelemetry.ExcludedUrls, ","),
+		})
+	}
+
+	if n.Spec.OpenTelemetry.DisableLogging != nil {
+		otelEnvVars = append(otelEnvVars, corev1.EnvVar{
+			Name:  "OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED",
+			Value: strconv.FormatBool(!*n.Spec.OpenTelemetry.DisableLogging),
+		})
+	} else {
+		otelEnvVars = append(otelEnvVars, corev1.EnvVar{
+			Name:  "OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED",
+			Value: strconv.FormatBool(true),
+		})
+	}
```

Comment:
nit: can be simplified
```suggestion
        var enableLog bool = true
	if n.Spec.OpenTelemetry.DisableLogging != nil {
	        enableLog = !*n.Spec.OpenTelemetry.DisableLogging
	}
        otelEnvVars = append(otelEnvVars, corev1.EnvVar{
		Name:  "OTEL_PYTHON_LOGGING_AUTO_INSTRUMENTATION_ENABLED",
		Value: strconv.FormatBool(enableLog),
	})
```

Reply:
done